toplevel/toploop.ml

```diff
@@ -117,11 +117,6 @@ let run_script ppf name args =
   let filename = filename_of_input name in
   Compmisc.init_path ~dir:(Filename.dirname filename) ();
```

Comment:
There is a bug hidden here: this PR is initializing the current environment before the call to `run_script`, which means that we are using the paths set up by `Topcommon.set_path ()` which adds the current directory rather than the directory of the script to the path.
Thus if we call
```bash
ocaml -open A inner/a.cmo inner/script.ml
```
with the following file available
```ocaml
- inner 
   | a.cmi
   | a.cmo
   | script.ml
 ```  
the interpreter fails with `unknown module A`.

Reply:
Good catch, I pushed a commit in this direction: https://github.com/ocaml/ocaml/pull/12888/commits/f2b69efb54358dd474edd85457f118b7d0a23c76